examples/blaze/src/main/scala/com/example/http4s/blaze/NestedResourceExample.scala

```diff
@@ -0,0 +1,31 @@
+package com.example.http4s.blaze
+
+import java.util.concurrent._
+import org.http4s._
+import org.http4s.client.Client
+import org.http4s.dsl._
+import org.http4s.server._
+import org.http4s.server.blaze._
+import org.http4s.client.blaze._
+import org.http4s.util.ProcessApp
+import scalaz._
+import scalaz.concurrent._
+import scalaz.stream._, Process._
+
+/** How to manage the lifecycle of a server and its dependent resources */
+object NestedResourceExample extends ProcessApp {
+  // This service depends on a client.
+  def service(client: Client): HttpService = HttpService {
+    case GET -> Root / "proxy" =>
+      client.toHttpService.run(Request(Method.GET, uri("http://http4s.org/")))
+  }
+
+  def main(args: List[String]) =
```

Comment:
I'm leaning toward this being a useful standalone library.  The downside is that there wouldn't be an instance for `Server` or `Client` without taking on a dependency or weird single-instance modules in one project or the other.